packages/uilib/src/lib/plugins/auto-doc/template-builder/template-builder.svelte

```diff
@@ -0,0 +1,83 @@
+<script lang="ts">
+
+  import ImageElement from "../elements/image/image-element.svelte";
+  import TextElement from "../elements/text/text-element.svelte";
+
+
+  let elements : Element[] = []
+
+  type Element = {
+    id: string,
+    type: "image" | "text" | "heading"
+    content: string | undefined
+  }
+
+
+  function generateRandomId(){
+    return Math.random().toString(16).slice(2)
+  }
+
+  function addElement(type: "image" | "text" | "heading"){
+    const newElement : Element = {
+      id: generateRandomId(),
+      type: type,
+      content: undefined
+    }
+
+    elements = [...elements, newElement];
+  }
+
+
+  function removeElement(id: string){
+    elements = elements.filter(e => e.id !== id);
+  }
+
+
+</script>
+
+<document-builder>
+
+  <div class="elements-container">
+    {#each elements as element (element.id)}
```

Comment:
Will refactor later into something else. For now the ifs and elses do the work

Reply:
The Idea is to create something like so:
<img width="589" alt="Bildschirmfoto 2024-10-18 um 12 15 18" src="https://github.com/user-attachments/assets/4480e002-2dc9-43a2-904b-e8d9222ee2fc">

However, I am not quite getting the correct binding and props communication to work out